capa/render/verbose.py

```diff
@@ -317,6 +318,58 @@ def render_rules(ostream, doc: rd.ResultDocument):
         ostream.writeln(rutils.bold("no capabilities found"))
 
 
+def render_domain_and_ip(ostream: rutils.StringIO, doc: rd.ResultDocument):
+    """
+    example::
+        +-----------------------------------------------------------+
+        | IP addresses and web domains                              |
+        |-----------------------------------------------------------+
+        | google.com                                                |
+        |    |----IP address:                                       |
+        |            |----192.0.0.1                                 |
+        |    |----Functions used to communicate with google.com:    |
+        |            |----InternetConnectA                          |
+        |            |----HttpOpenRequestA                          |
+        |            |----FtpGetFileA                               |
+        |    |----3 occurrances                                     |
+        |                                                           |                                                                          |
+        | 192.123.232.08                                            |
+        |    |----Functions used to communicate with 192.123.232.08:|
+        |            |----...                                       |
+        |                                                           |
+        +-----------------------------------------------------------+
+    """
+    rows = []
+    for domain_or_ip in verbose_extract_domain_and_ip(doc):
+        for i in domain_or_ip.split("\n"):
+            rows.append(i)
+
+    max_line = 0
+    for item in rows:
+        for new_line in item.split("\n"):
+            if len(new_line) > max_line:
+                max_line = len(new_line)
+
+    white_spaces = " " * ceil(1 / 3 * max_line)
+
+    if rows:
+        ostream.write(
+            tabulate.tabulate(
+                {white_spaces + "IP addresses and web domains" + white_spaces: rows},
+                headers=[white_spaces + "IP addresses and web domains" + white_spaces],
+                tablefmt="mixed_outline",
+            )
+        )
+        ostream.write("\n")
+    else:
+        ostream.writeln(rutils.bold("No web domains or IP addresses found"))
+
+
+def ceil(num):
+    if isinstance(num, float):
```

Comment:
what if `num` is integer? The return is `None`. Do you want this behaviour?

but there is a function in the math default package of python which do that not?

Reply:
This is a good question - it shouldn't be an issue because the input to ```ceil``` will always be some multiple of 1/3, so the input's type should always be float, but you're right that it's not explicit - I will add a type hint here. I thought it might be a bit unnecessary to import the  math package just for a simple function lol but what do you think?

Reply:
Just updated, please let me know what you think!

Reply:
Now the method is better :)

I have a problem with the fact that you implement the same method in two different scripts. If you have a bug in one method you have the same bug in the other, i.e, two bugs.

However, it is improbable that you have a bug in this method ;)

Using math.ceil is not a problem, math is a standard library in python.

But up to you ;)